plugins/colorbutton/plugin.js

```diff
@@ -364,6 +405,22 @@ CKEDITOR.plugins.add( 'colorbutton', {
 			// Replace 3-character hexadecimal notation with a 6-character hexadecimal notation (#1008).
 			return CKEDITOR.tools.normalizeHex( '#' + CKEDITOR.tools.convertRgbToHex( color || '' ) ).replace( /#/g, '' );
 		}
+
+		function _registerStateSynchronization( command, uiElement ) {
+			command.on( 'state', function() {
+				if ( this.state !== uiElement.getState() ) {
+					uiElement.setState( this.state );
+				}
+			} );
+
+			uiElement.setState = function( state ) {
```

Comment:
The reverse order of updating the state seems unnecessary. You can test it with other buttons in CKEditor, e.g.:

```javascript
var button = editor.ui.get( 'Bold' );
button.setState( 0 );
var command = editor.getCommand( 'bold' );
command.state; // => 2
```

It may seem like they should synchronize, but because we are not doing it in our codebase, I recommend to stick to adopted logic.